src/Core/Utilities/ServiceContainer.cs

```diff
@@ -129,6 +129,59 @@ public static T Resolve<T>(string serviceName, bool dontThrow = false)
             throw new Exception($"Service {serviceName} is not registered.");
         }
 
+        public static void Register<T>(T obj)
+            where T : class
+        {
+            Register(typeof(T), obj);
+        }
+
+        public static void Register(Type type, object obj)
+        {
+            var serviceName = GetServiceRegistrationName(type);
+            if (!RegisteredServices.TryAdd(serviceName, obj))
+            {
+                throw new Exception($"Service {serviceName} has already been registered.");
```

Comment:
Do we really need to throw? We could just register the new one and discard the previous reference.

Reply:
I wanted not to change much of the previous logic, but yes definitely we can do that and it's the way most of the DI engines work. And today it's throwing an exception so if nothing is currently broken we can overwrite the registration without any possible drawbacks